qiskit/backends/ibmq/ibmqprovider.py

```diff
@@ -19,19 +20,24 @@ def __init__(self, token, url='https://quantumexperience.ng.bluemix.net/api',
                  hub=None, group=None, project=None, proxies=None, verify=True):
         super().__init__()
 
+        if any([hub, group, project]):
```

Comment:
Seeing that this block of code is used in several places - can you move it to a separate function, in a place where it is easy to import (for example, `qiskit._utils.parse_ibmq_credentials(url, hub, group, project)`  - a better name might be needed)? This would avoid duplication, and we can take care of the checking, transforming and deprecation raising in a single place.

Reply:
Done.